src/can-deploy.ts

```diff
@@ -97,22 +100,28 @@ export class CanDeploy {
     instance.stdout.on('data', l => output.push(l));
     instance.stderr.on('data', l => output.push(l));
     instance.once('close', code => {
-      const o = output.join('\n');
+      const result: any = output.join('\n');
 
-      let success = false;
       if (this.options.output === 'json') {
-        success = JSON.parse(o).summary.deployable;
-      } else {
-        success = /Computer says yes/gim.exec(o) !== null;
+        try {
+          const parsed = JSON.parse(result);
+          if (code === 0 && parsed.summary.deployable) {
+            return deferred.resolve(parsed);
+          }
+          return deferred.reject(parsed);
+        } catch (e) {
+          logger.error(`can-i-deploy produced non-json output:\n${result}`);
+          return deferred.reject(new Error(result));
+        }
       }
 
-      if (code === 0 || success) {
-        logger.info(o);
-        return deferred.resolve();
+      if (code === 0) {
+        logger.info(result);
+        return deferred.resolve(result);
       }
 
-      logger.error(`can-i-deploy did not return success message:\n${o}`);
-      return deferred.reject(new Error(o));
+      logger.error(`can-i-deploy did not return success message:\n${result}`);
+      return deferred.reject(result);
```

Comment:
Are we ok that this will not distinguish between deployment failure vs rejecting with an `Error` on binary call error?

Reply:
Good call, see above union type update.